dali/operators/decoder/audio/audio_decoder.h

```diff
@@ -48,7 +48,9 @@ class AudioDecoderBase {
    */
   virtual ptrdiff_t Decode(span<char> raw_output) = 0;
 
-  virtual ~AudioDecoderBase() = default;
+  virtual ~AudioDecoderBase() {
+    Close();
```

Comment:
It looks like it can't work. At the time you reach `~AudioDecoderBase` the derived class is destroyed and `CloseImpl` would point to a pure virtual function.

Reply:
Fixed